File: scala/src/main/scala/cn/spark/core/BroadcastVariable.scala
package scala.cn.spark.core

import org.apache.spark.{SparkConf, SparkContext}

/**
  * Created by wangzhilong on 2017/6/5.
  */
object BroadcastVariable {

  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
      .setAppName("BroadcastVariable")
      .setMaster("local")
    val sc = new SparkContext(conf)
    val factor = sc.broadcast(3);
    sc.parallelize(Array(1,2,3,4,5))
      .map(num => num * factor.value)
      .foreach(println(_))
  }
}

File: java/src/main/java/cn/hadoop/core/BroadcastVariable.java
package cn.hadoop.core;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.broadcast.Broadcast;

import java.util.Arrays;
import java.util.List;

/**
 * Created by wangzhilong on 2017/6/5.
 */
public class BroadcastVariable {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf()
                .setAppName("BroadcastVariable")
                .setMaster("local");
        JavaSparkContext sc = new JavaSparkContext(conf);

        final int factor = 3 ;
        final Broadcast<Integer> factorBroadcast = sc.broadcast(factor);

        List<Integer> numberList = Arrays.asList(1,2,3,4,5) ;
        JavaRDD<Integer> numbers = sc.parallelize(numberList);
        numbers.map(n -> n * factorBroadcast.value())
                .foreach(n -> System.out.println(n));
        sc.close();
    }
}
